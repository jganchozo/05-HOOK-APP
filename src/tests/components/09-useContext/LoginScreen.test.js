import { mount } from 'enzyme';
import React from 'react';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';


describe('Pruebas en LoginScreen', () => { 
    
    const setUser = jest.fn();

    const wrapper = mount(
        <UserContext.Provider value={{
            setUser
        }}>
            <LoginScreen /> );
        </UserContext.Provider>
    );

    test('should devolver el componente correcto', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should llamar el setUser con el click', () => {

        //wrapper.find('button').simulate('click');
        wrapper.find('button').prop('onClick')();
        expect(setUser).toHaveBeenCalledWith({
            id: 12345,
            name: 'Jose Ganchozo'
        });
    });

});