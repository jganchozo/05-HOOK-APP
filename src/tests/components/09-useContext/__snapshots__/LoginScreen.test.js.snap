// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en LoginScreen should devolver el componente correcto 1`] = `
Array [
  <div>
    <h1>
      LoginScreen
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>,
  " );",
]
`;
